react/Payment.tsx

```diff
@@ -1,9 +1,65 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { useOrderPayment } from 'vtex.order-payment/OrderPayment'
+import { Router } from 'vtex.checkout-container'
 
-import CreditCard from './components/CreditCard'
+import CreditCard from './CreditCard'
+import Installments from './Installments'
+import PaymentList from './PaymentList'
+import { PaymentStage } from './enums/PaymentEnums'
 
+const REVIEW_ROUTE = '/'
 const Payment: React.FC = () => {
-  return <CreditCard />
+  const [stage, setStage] = useState<PaymentStage>(PaymentStage.PAYMENT_LIST)
+  const { cardFormData, setCardFormData, setPaymentField } = useOrderPayment()
+  const history = Router.useHistory()
+
+  const onCardFormCompleted = () => {
+    setStage(PaymentStage.INSTALLMENTS)
+  }
+
+  const onInstallmentSelected = (installment: number) => {
+    setPaymentField({
+      installments: installment,
+    })
+    history.push(REVIEW_ROUTE)
+  }
+
+  const backToCreditCard = () => {
+    setStage(PaymentStage.CARD_FORM)
+  }
+
+  const newCreditCard = () => {
+    setCardFormData(null)
+    setStage(PaymentStage.CARD_FORM)
+  }
+
+  const editCard = () => {
+    setStage(PaymentStage.CARD_FORM)
+  }
+
+  const backToPaymentList = () => {
+    setStage(PaymentStage.PAYMENT_LIST)
+  }
+
+  return (
+    <>
+      <div className={stage === PaymentStage.CARD_FORM ? '' : 'dn'}>
```

Comment:
Did you think of `{stage === PaymentStage.CARD_FORM && <div className="">...</div>`? Might be lighter than this DOM manipulation.

Reply:
actually, this is needed for the `iframe` to be "mounted" all the time, else when you change route, say, from credit card form to installments, and go back, the information on the card form will be lost